content/webapp/components/CalendarSelect/NewCalendarSelect.tsx

```diff
@@ -0,0 +1,53 @@
+import { FunctionComponent } from 'react';
+import Select from '@weco/content/components/Select';
+import { dateAsValue } from '../ItemRequestModal/format-date';
+import { AvailabilitySlot } from '@weco/content/services/wellcome/catalogue/types';
+import { formatDayName, formatDayMonth } from '@weco/common/utils/format-date';
+
+type Props = {
+  availableDates: AvailabilitySlot[];
+  chosenDate?: string;
+  setChosenDate: (value: string) => void;
+};
+
+const availabilitySlotsToSelectOptions = (
+  availableDates: AvailabilitySlot[]
+) => {
+  // AvailabilitySlots have open and close dateTimestamps
+  // right now we only care about the day, not the time
+  // so we're only using "from" = opening date/time
+  return (
+    availableDates
+      .map(availabilitySlot => new Date(availabilitySlot.from))
+      .map(availableDate => ({
+        value: dateAsValue(availableDate),
+        text: `${formatDayName(availableDate)} ${formatDayMonth(
+          availableDate
+        )}`,
+      }))
+      // the list of available dates is returned from the itemsAPI in various lenghts
+      // trimming down to 12 for consistency
+      .slice(0, 12)
+  );
+};
+
+const NewCalendarSelect: FunctionComponent<Props> = ({
+  availableDates,
+  chosenDate,
+  setChosenDate,
+}) => {
+  return availableDates.length ? (
+    <Select
+      name="calendar_dates"
+      label="Select a date"
+      hideLabel={true}
+      options={availabilitySlotsToSelectOptions(availableDates)}
+      value={chosenDate || 'Select a date'}
+      onChange={e => setChosenDate(e.target.value)}
+    />
+  ) : (
+    <>Error: Available dates could not be found.</>
```

Comment:
In case there is an error in the itemsAPI when it fetches the opening times from the contentAPI, the itemsAPI returns an empty list of dates for that item, rather than than failing the entire request.
This means there is a small, yet non-null possibility that the list of available dates will be empty for one/all of the work's items. I'm choosing to wait for the last step to surface this error because it doesn't mean the item is not requestable, we just couldn't get the dates.
Is this the right way to go, and if it is, how can we make it look better (this is more like a placeholder, looks pretty terrible on the page)

Reply:
Does this need a designer or can we figure it out on a call tomorrow maybe?

Reply:
Copy requested in Slack https://wellcome.slack.com/archives/C3TQSF63C/p1715246147417789